react-carousel-component/src/carousel.jsx

```diff
@@ -0,0 +1,149 @@
+    const alt = this.props.alt;
+
+    return (
+      <div className="image-container"><img src={address} alt={alt} /></div>
+    );
+  }
+}
+
+class DotIconLocation extends React.Component {
+  render() {
+    const location = this.props.id;
+    const result = [];
+
+    for (let i = 0; i < 5; i++) {
+      if (i !== location) {
+
+        result.push(<Tag dot={dots.whiteDot} indexOfImage={i.toString()}/>);
+      } else {
+        result.push(<Tag dot={dots.blackDot} indexOfImage={i.toString()}/>);
+      }
+    }
+
+    return (
+    <div className="page-icon-container">
+      {result}
+    </div>
+
+    );
+  }
+
+}
+
+class Carousel extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      activeInterval: null,
+      currentImgInfo: {},
+      currentIndex: 0,
+      isClickedChangeImageIcon: false
+    };
+
+    this.imageChange = this.imageChange.bind(this);
+    this.handleClick = this.handleClick.bind(this);
+    this.resumePlay = this.resumePlay.bind(this);
+  }
+
+  imageChange() {
+    const images = this.props.puppyImages;
+    let index = this.state.currentIndex;
+
+    this.setState({ currentImgInfo: images[index] });
+
+    const intervalId = setInterval(() => {
+      index++;
+      if (index === 5) {
+        index = 0;
+      }
+      this.setState({ currentImgInfo: images[index] });
+      this.setState({ currentIndex: index });
+    }
+    , 3000);
+    this.setState({ activeInterval: intervalId });
+
+  }
+
+  handleClick(event) {
+    clearInterval(this.state.activeInterval);
+    this.setState({ activeInterval: null });
+    this.setState({ isClickedChangeImageIcon: true });
+
+    const classLists = event.target.classList;
+    if (event.target.tagName === 'I') {
+      for (let i = 0; i < classLists.length; i++) {
+        if (classLists[i] === 'left-arrow') {
+          let previousImgIndex = this.state.currentIndex - 1;
+          if (previousImgIndex === -1) {
+            previousImgIndex = 4;
+          }
+          this.setState({ currentIndex: previousImgIndex });
+        } else if (classLists[i] === 'right-arrow') {
+          let nextImgIndex = this.state.currentIndex + 1;
+          if (nextImgIndex === 5) {
+            nextImgIndex = 0;
+          }
+          this.setState({ currentIndex: nextImgIndex });
+        } else {
+          const getIndexFromDot = parseInt(classLists[2]);
+          this.setState({ currentIndex: getIndexFromDot });
+        }
+      }
+      const images = this.props.puppyImages;
+      const index = this.state.currentIndex;
+
+      this.setState({ currentImgInfo: images[index] });
+
+    }
+
+  }
+
+  resumePlay() {
+    this.setState({ isClickedChangeImageIcon: false });
+  }
+
+  render() {
+    if (!this.state.activeInterval && !this.state.isClickedChangeImageIcon) {
+      this.imageChange();
+    } else if (this.state.isClickedChangeImageIcon) {
+      setTimeout(() => {
+        this.resumePlay();
+      }, 3000);
+    }
```

Comment:
Calling any method in render that changes the state is dangerous, you can create infinite loops. It would be better to start the interval in [componentDidMount](https://reactjs.org/docs/state-and-lifecycle.html#adding-lifecycle-methods-to-a-class). Then in your handle click you can clear the interval at the start of the method then after you do everything you need to do you can restart the interval at the end of the method.